src/Language/Jsonnet/Common.hs

```diff
@@ -65,7 +65,7 @@ instance Alpha UnyOp
 data KeyValue a = KeyValue
   { key :: a,
     value :: a
-  }
+  } | Hidden { key :: a }
```

Comment:
Hidden fields, also have values, e.g.
```hs
{ foo : "bar", baz(x) :: x + 1 }
```
In this case, you might use the baz method in the program, regardless of the object being serialized to JSON.

Another example, the functions in the standard library are all hidden fields of the std object (https://jsonnet.org/ref/stdlib.html)

Reply:
My suggestion here would be to add an additional field `hidden :: Bool`.